### PR TITLE
Allow building with LLVM 8

### PR DESCRIPTION
**Pull request by mppf**
* updates update-llvm.sh to checkout the official LLVM git monorepo
* updates Makefiles to use llvm-project (created from the above) instead of llvm-src, if it exists
* updates compiler sources to build with LLVM 8.

- [x] full local testing
- [x] full local --llvm testing

Reviewed by @dmk42 - thanks!

### CODE REVIEW

**Comment by mppf** [comment]
@dmk42 - FYI. This is not urgent, just wanted to let you know I've created it.

**Comment by mppf** [comment]
@dmk42 - note that in the future, I expect we'll move from llvm-src to llvm-project. When that happens, we'll want to rename the checked-out-LLVM-trunk directory to something else, e.g. llvm-project-clone.

File: compiler/llvm/clangUtil.cpp [hunk 1]
```diff
@@ -2469,7 +2469,11 @@ void setupForGlobalToWide(void) {
 #if HAVE_LLVM_VER < 50
                           , NULL
 #endif
+#if HAVE_LLVM_VER < 80
                           );
+#else
+                          ).getCallee();
```
Comment (dmk42) [inline compiler/llvm/clangUtil.cpp]: @mppf - I just updated to master and found that this code does not compile for LLVM 8.  The `.getCallee()` should be `->getCallee()` and similarly with the other .function() calls.  I get the same failure on Mac Homebrew LLVM 8 and Ubuntu LLVM 8.

Comment (mppf) [reply compiler/llvm/clangUtil.cpp]: It needs to be .getCallee for LLVM 9 (pre-release). I havn't been able to reproduce your issue. Can you prepare a PR?

Comment (mppf) [reply compiler/llvm/clangUtil.cpp]: (I havn't tried to build with LLVM 8 myself yet but it's probably faster for you to fix than for me to get that going)

**Pull request merged**